controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -143,6 +144,20 @@ public void initializeStreamWriters(final EventStreamClientFactory clientFactory
         writerInitFuture.complete(null);
     }
 
+    /**
+     * Method to create stream with retries for lock failed exception.
+     *
+     * @param scope           scope.
+     * @param stream          stream name.
+     * @param config          stream configuration.
+     * @param createTimestamp creation timestamp.
+     * @return CompletableFuture which when completed will have creation status for the stream.
+     */
+    public CompletableFuture<CreateStreamStatus.Status> createStreamWithReries(String scope, String stream, StreamConfiguration config, long createTimestamp) {
+        return RetryHelper.withRetriesAsync(() ->  createStream(scope, stream, config, createTimestamp), 
+                e -> Exceptions.unwrap(e) instanceof LockFailedException, 10, executor);
```

Comment:
In the case that the number of retries exhausts, what's the error that the application will see? Is it `RetriesExhaustedException`? I'd rather have the result of the last attempt, which should be more informative.

Reply:
Currently it would be RetriesExhausted that wraps LockFailedException. 
I will unwrap it.